common/src/main/java/io/netty/util/concurrent/SingleThreadEventExecutor.java

```diff
@@ -811,15 +802,11 @@ protected boolean confirmShutdown() {
 
     @Override
     public boolean awaitTermination(long timeout, TimeUnit unit) throws InterruptedException {
-        if (unit == null) {
-            throw new NullPointerException("unit");
-        }
-
         if (inEventLoop()) {
             throw new IllegalStateException("cannot await termination of the current thread");
         }
 
-        threadLock.await(timeout, unit);
+        threadLock.await(timeout, ObjectUtil.checkNotNull(unit, "unit"));
```

Comment:
move check back as first in the method to not change behaviour if null is passed as `unit` but we are in the `EventLoop` thread.